Apps/phpanorama/panorama_connector.py

```diff
@@ -898,15 +923,72 @@ def _block_url(self, param):
 
         action_result = self.add_action_result(ActionResult(dict(param)))
 
+        status = self._load_pan_version(action_result)
+        if phantom.is_fail(status):
+            return action_result.set_status(
+                phantom.APP_ERROR, PAN_ERR_MSG.format("blocking url", action_result.get_message()))
+
+        # Pick BlockUrl handlers based on the major version of Panorama.
+        major_version = self._get_pan_major_version()
+        if major_version < 9:
+            return self._block_url_8_and_below(param, action_result)
+
+        return self._block_url_9_and_above(param, action_result)
+
+    def _block_url_9_and_above(self, param, action_result):
         if param['policy_type'] not in POLICY_TYPE_VALUE_LIST:
             return action_result.set_status(phantom.APP_ERROR, VALUE_LIST_VALIDATION_MSG.format(POLICY_TYPE_VALUE_LIST, 'policy_type'))
 
-        # Check if policy is present or not
         status, policy_present = self._does_policy_exist(param, action_result)
         action_result.set_data_size(0)
         if phantom.is_fail(status):
             return action_result.set_status(phantom.APP_ERROR, PAN_ERR_MSG.format("blocking url", action_result.get_message()))
 
+        if not policy_present:
+            return action_result.set_status(
+                phantom.APP_ERROR, PAN_ERR_POLICY_NOT_PRESENT_CONFIG_DONT_CREATE)
+
+        url_prof_name = BLOCK_URL_PROF_NAME.format(
+            device_group=self._handle_py_ver_compat_for_input_str(param[PAN_JSON_DEVICE_GRP]))
+        url_prof_name = url_prof_name[:MAX_NODE_NAME_LEN].strip()
+
+        status = self._create_or_update_url_category(param, action_result, url_prof_name)
+        if phantom.is_fail(status):
+            error_msg = PAN_ERR_MSG.format("blocking url", action_result.get_message())
+            return action_result.set_status(phantom.APP_ERROR, error_msg)
+
+        status = self._create_or_update_url_filtering(param, action_result, url_prof_name)
+        if phantom.is_fail(status):
+            error_msg = PAN_ERR_MSG.format("blocking url", action_result.get_message())
+            return action_result.set_status(phantom.APP_ERROR, error_msg)
```

Comment:
Here, we modify the 1st step of BlockUrl. The remaining steps work the same as before.